app/controllers/application_controller.rb

```diff
@@ -252,26 +252,24 @@ def process_success(hash = {})
 
   def process_error(hash = {})
     hash[:object] ||= instance_variable_get("@#{controller_name.singularize}")
-
-    case action_name
-    when "create" then hash[:render] ||= "new"
-    when "update" then hash[:render] ||= "edit"
-    else
-      hash[:redirect] ||= send("#{controller_name}_url")
+    if hash[:render].blank? && hash[:redirect].blank?
+      case action_name
+      when "create" then hash[:render] ||= "new"
+      when "update" then hash[:render] ||= "edit"
+      else
+        hash[:redirect] ||= send("#{controller_name}_url")
```

Comment:
nit - `hash[:redirect]` can't be blank, same applies to `hash[:render]` in both when branches so `||=` should be just `=`